### PR TITLE
Add forward pass test for Llama2-7b

### PR DESCRIPTION
**Pull request by A9isha**

### CODE REVIEW

File: MaxText/tests/fwd_pass_testutil.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import numpy as np
+
+import pyconfig
+
+import jsonlines
+import train
+
+
+def get_data(golden_data, golden_data_index, config):
+    print(f"Comparing forward pass for golden data index = {golden_data_index} ")
```
Comment (rwitten) [inline MaxText/tests/fwd_pass_testutil.py:38]: probably stick to maxlogging

File: MaxText/tests/fwd_pass_testutil.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      state_mesh_annotations,
+      model,
+      mesh,
+      learning_rate_schedule,
+      data_iterator,
+      eval_data_iterator,
+      state,
+    ) = train.setup_train_loop(config)
+  
+  input_golden_data_path = "MaxText/test_assets/golden_data_"+config.model_name+".jsonl"
```
Comment (rwitten) [inline MaxText/tests/fwd_pass_testutil.py:76]: probably worth exposing directly?

File: end_to_end/tpu/llama2/7b/test_llama2_7b.sh [hunk 3]
```diff
@@ -71,3 +71,6 @@ export NEW_CKPT_PATH=${BASE_OUTPUT_DIRECTORY}/${PARAMETER_CHECKPOINT_RUN}/checkp
 
 # We run decoding on the fine-tuned parameter checkpoint
 python3 MaxText/decode.py MaxText/configs/base.yml load_parameters_path=${NEW_CKPT_PATH} run_name=runner_decode_finetuned_${idx} base_output_directory=${BASE_OUTPUT_DIRECTORY} per_device_batch_size=1 model_name='llama2-7b' ici_autoregressive_parallelism=4 max_prefill_predict_length=4  max_target_length=16 prompt="I love to" attention=dot_product scan_layers=false
+
+# We also test whether the forward pass logits match the golden logits for Llama2-7b
+python3 MaxText/tests/fwd_pass_testutil.py  MaxText/configs/base.yml load_parameters_path=${UNSCANNED_CKPT_PATH} run_name=forward_pass_test per_device_batch_size=1 model_name=llama2-7b ici_tensor_parallelism=4 max_prefill_predict_length=4 max_target_length=4 dataset_type=synthetic  dtype=float32 scan_layers=false
```
Comment (rwitten) [inline end_to_end/tpu/llama2/7b/test_llama2_7b.sh:76]: probably we should pass the prompt as an argument? just consider if the default prompt changes.

File: MaxText/tests/fwd_pass_testutil.py (hunk 2 above)
Comment (rwitten) [reply MaxText/tests/fwd_pass_testutil.py:76]: like kind of weird to be guessing -- what if the someone wants to test multiple prompts, etc.

**Review by rwitten — CHANGES_REQUESTED** [review]
Broadly like it! I think the standalone test file is great but not sure I like the name. Maybe `forward_pass_logit_checker.py`

File: MaxText/tests/fwd_pass_testutil.py (hunk 2 above)
Comment (A9isha) [reply MaxText/tests/fwd_pass_testutil.py:76]: >probably worth exposing directly?

Since we are comparing logts, for Llama2 we are looking for `[sequence_length,32000]` long logits, it would be hard to expose directly in this `.py` file itself.

This input `.jsonl` file has multiple prompts already, three actually, users can add more.

Is this what you meant?

File: end_to_end/tpu/llama2/7b/test_llama2_7b.sh (hunk 3 above)
Comment (A9isha) [reply end_to_end/tpu/llama2/7b/test_llama2_7b.sh:76]: The prompts and the golden logits are all rolled in the  input `jsonl` file under `test_assets`. 

Currently the 3 prompts for this logit checker script are `I love to`, `Today is a`, `What is the`

We are not computing the golden logits live during the test, we are keep the prompts+golden logits saved and only running MaxText logits during the test.

File: MaxText/tests/fwd_pass_testutil.py (hunk 1 above)
Comment (A9isha) [reply MaxText/tests/fwd_pass_testutil.py]: done - thanks!

File: MaxText/test_assets/golden_data_llama2-7b.jsonl
[the remark this answers is not in the document]
Comment (A9isha) [reply MaxText/test_assets/golden_data_llama2-7b.jsonl:1]: Great point Matt - I have added the documentation in MaxText/tests/forward_pass_logit_checker.py

**Review by rwitten — APPROVED** [review]
